### PR TITLE
Add data wrangling glossary entry

### PR DESCRIPTION
**Pull request by johnblust**
## Description & motivation
Adds data wrangling glossary entry to the docs repo

### CODE REVIEW

File: website/docs/terms/data-wrangling.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+### Structuring
+
+Structuring your data is a type of transformation that involves reformatting and reorganizing your data so that it is stored in a way that makes the values usable. This could mean rearranging how the data is displayed in columns and rows. Chances are you are using an <Term id="etl" /> tool to ingest your data, so the data is likely in a tabular format and you won’t need to do that much restructuring. If your data is structured, you really only need to worry about nested data types such as <Term id="json" /> data. When structuring your data, you want to ask yourself these questions:
+
+- Is your data in the format you need to perform analysis on it? Does your data potentially need to be unnested? *Should you nest or objectize columns together?*
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- Is your data in the format you need to perform analysis on it? Does your data need to be potentially unnested? *Should you nest or objectize columns together?*
- Do the column names and values look correct for your use case?
```

File: website/docs/terms/data-wrangling.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Data discovery involves getting to know the data that you are working with. This involves looking at key statistical measures of your dataset. Some of these include:
+
+- Row count
+- Number of columns
+- Column data types
+- Distribution of column values
+- Number of duplicate rows
+- Number of nulls
+
+Oftentimes, data warehouses have a preview capability so data team members can easily see a table’s makeup (column name, type, row count, etc.), but functions such as SUM() and COUNT() will come in handy for finding these values. You can use the GROUP BY statement with these functions to find the counts of certain rows for different categories of data. In addition, you’ll want to identify primary keys, check for duplicates of primary keys, and ensure every row of data has a column that can act as a primary key!
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
Oftentimes, data warehouses have a preview capability so data team members can easily see a table’s makeup (column name, type, row count, etc.), but functions such as `SUM()` and `COUNT()` will come in handy for finding these values. You can use the `GROUP BY` statement with these functions to find the counts of certain rows for different categories of data. In addition, you’ll want to identify <Term id = "primary-key">primary keys</Term>, check for duplicates of primary keys, and ensure every row of data has a column that can act as a primary key!
```

File: website/docs/terms/data-wrangling.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- Eliminating unnecessary characters or spaces from values
+
+Certain cleaning steps, like removing rows with null values, are helpful to do at the beginning of the process because removing nulls and duplicates from the start can increase the performance of your downstream models.  In the cleaning step, it’s important to follow a standard for your transformations here. This means you should be following a consistent naming convention for your columns (especially for your <Term id="primary-key">primary keys</Term>) and casting to the same timezone and datatypes throughout your models. Examples include making sure all dates are in UTC time rather than source timezone-specific, all string in either lower or upper case, etc. 
+
+:::tip dbt to the rescue!
+If you are struggling to do all the cleaning on your own, remember that dbt packages ([dbt expectations](https://github.com/calogica/dbt-expectations), [dbt_utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/), and [re_data](https://www.getre.io/)) and their macros are also available to help you clean values if they can’t be cleaned with simple functions.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
If you're struggling to do all the cleaning on your own, remember that dbt packages ([dbt expectations](https://github.com/calogica/dbt-expectations), [dbt_utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/), and [re_data](https://www.getre.io/)) and their macros are also available to help you clean up your data.
```

File: website/docs/terms/data-wrangling.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+:::
+
+### Enriching
+
+Enriching your data means enhancing it by supplementing incomplete or missing data. This could involve basic case or coalesce statements that use an already existing column in order to produce a new column. It could also look like joining an already existing date column with a date table that contains more extensive information about a certain date. Keep in mind that you don’t want to go overboard with enriching or joining here—you only want to add what will be repeatedly used in modeling and analysis.
+
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md:64]:
```suggestion
:::tip Python for enrichment?
With the new capability of [[Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models)](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [[#beta-feedback-python-models channel](https://www.getdbt.com/community/join-the-community/)](https://www.getdbt.com/community/join-the-community/) and join in the [[GitHub discussions](https://github.com/dbt-labs/dbt-core/discussions/5261)](https://github.com/dbt-labs/dbt-core/discussions/5261) happening around it.
:::

```

File: website/docs/terms/data-wrangling.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+### Standardization
+
+When data is wrangled, it is matched with a standard set that your data team establishes that is then applied to all datasets. It ultimately creates consistent staging layers for analytics engineers to build their intermediate, fct/dim, and mart models. Data team members don’t need to worry about upholding standards in downstream models because this is already done when the data is first ingested.
+
+### Deeper understanding of data
+
+By first wrangling or cleaning data, you get to learn about the data’s intricacies in the process. Though manual, this process allows you to find issues in the data and understand them deeply before using them in downstream processes. This minimzes potential problems that can go unnoticed because you’ve already explored and validated the datasets. It also helps you understand how tables can be joined together downstream.
+
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md:114]:
```suggestion
Additionally, this initial data exploration and transformation helps you collaborate better with [backend application developers](https://docs.getdbt.com/blog/when-backend-devs-spark-joy) or data engineers to work on formatting the raw data in a format that is most appropriate for analytics work.

### Potential performance improvements on downstream models

Lastly, data wrangling allows for potential improvements in performance in downstream models. Because you’ve cleaned the data and potentially removed duplicates and null values, models will be quicker to run.
```

File: website/docs/terms/data-wrangling.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+## Data wrangling in SQL
+
+SQL is the most common language for data wrangling. While you can wrangle data using other languages, such as Python, SQL is the most common (and straightforward) language used for data wrangling and transformation in relational databases. Let’s look at some of the most common SQL functions for each of the data wrangling steps. 
+
+### SQL cleaning
+
+CAST( <column_name> AS date) is commonly used to cast values in a column to a date datatype, or whatever other datatype you specify.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `CAST( <column_name> AS date)` is commonly used to cast values in a column to a date datatype, or whatever other datatype you specify.
```

File: website/docs/terms/data-wrangling.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+CONVERT_TZ() can be used to convert values in a column to a specific timezone.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `CONVERT_TZ()` can be used to convert values in a column to a specific timezone.
```

File: website/docs/terms/data-wrangling.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+LOWER()/UPPER() is used to uncaptialize strings.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `LOWER()`/`UPPER() `is used to capitalize or lowercase string values.
```

File: website/docs/terms/data-wrangling.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+TRIM() eliminates white spaces in strings, making string functions easier to use downstream.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
`TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```

File: website/docs/terms/data-wrangling.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+REPLACE() replaces a specified character in column values.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
`REPLACE()` replaces a specified character in column values.
```

File: website/docs/terms/data-wrangling.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+You can also use custom built macros, such as those from a free dbt package called re_data, to clean columns using SQL.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
You can also use custom built macros, such as those from a free dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL. 
```

File: website/docs/terms/data-wrangling.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+TRIM() eliminates white spaces in strings, making string functions easier to use downstream.
+
+REPLACE() replaces a specified character in column values.
+
+You can also use custom built macros, such as those from a free dbt package called re_data, to clean columns using SQL. 
+
+### Enriching
+
+Enriching data using SQL often involves the following three functions:
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
Enriching data using SQL can often involve the use of functions, such as:
```

File: website/docs/terms/data-wrangling.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements. 
```

File: website/docs/terms/data-wrangling.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+REPLACE() replaces a specified character in column values.
+
+You can also use custom built macros, such as those from a free dbt package called re_data, to clean columns using SQL. 
+
+### Enriching
+
+Enriching data using SQL often involves the following three functions:
+
+CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements. 
+
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
```

File: website/docs/terms/data-wrangling.md [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+IFNULL() replaces any NULL values in a column with whatever value you specify.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
`IFNULL()` replaces any NULL values in a column with whatever value you specify.
```

File: website/docs/terms/data-wrangling.md [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
```

File: website/docs/terms/data-wrangling.md [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+COALESCE() returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
`COALESCE()` returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```

File: website/docs/terms/data-wrangling.md [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md:141]:
```suggestion
```

File: website/docs/terms/data-wrangling.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+### Structuring
+
+Pivot tables come in handy when restructuring your data. You can use them to make your column names your values and vice versa. Dbt has a [macro](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/pivot.sql) built out that allows you to completely customize and pivot your tables without having to write crazy complicated code.
+
+For nested data types such as JSON, you’ll want to check out the JSON parsing and extraction function of your data warehouse to help work with this data.
+
+### Validating
+
+In addition, dbt offers [generic tests](https://docs.getdbt.com/docs/building-a-dbt-project/tests#more-generic-tests) in every dbt project that allows you to validate accepted, unique, and null values. They also allow you to validate the relationships between tables and that the primary key is unique.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
dbt offers [generic tests](https://docs.getdbt.com/docs/building-a-dbt-project/tests#more-generic-tests) in every dbt project that allows you to validate accepted, unique, and null values. They also allow you to validate the relationships between tables and that the primary key is unique.
```

File: website/docs/terms/data-wrangling.md [hunk 20]
```diff
@@ -0,0 +1,157 @@
...
+
+:::tip dbt to the rescue!
+If you're struggling to do all the cleaning on your own, remember that dbt packages ([dbt expectations](https://github.com/calogica/dbt-expectations), [dbt_utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/), and [re_data](https://www.getre.io/)) and their macros are also available to help you clean up your data.
+:::
+
+### Enriching
+
+Enriching your data means enhancing it by supplementing incomplete or missing data. This could involve basic case or coalesce statements that use an already existing column in order to produce a new column. It could also look like joining an already existing date column with a date table that contains more extensive information about a certain date. Keep in mind that you don’t want to go overboard with enriching or joining here—you only want to add what will be repeatedly used in modeling and analysis.
+:::tip Python for enrichment?
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md:65]:
```suggestion

:::tip Python for enrichment?
```

File: website/docs/terms/data-wrangling.md [hunk 21]
```diff
@@ -0,0 +1,157 @@
...
+With the new capability of [[Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models)](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [[#beta-feedback-python-models channel](https://www.getdbt.com/community/join-the-community/)](https://www.getdbt.com/community/join-the-community/) and join in the [[GitHub discussions](https://github.com/dbt-labs/dbt-core/discussions/5261)](https://github.com/dbt-labs/dbt-core/discussions/5261) happening around it.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
With the new capability of [Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [#beta-feedback-python-models channel](https://www.getdbt.com/community/join-the-community/) and join in the [GitHub discussions](https://github.com/dbt-labs/dbt-core/discussions/5261) happening around it.
```

File: website/docs/terms/data-wrangling.md [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+
+### Standardization
+
+When data is wrangled, it is matched with a standard set that your data team establishes that is then applied to all datasets. It ultimately creates consistent staging layers for analytics engineers to build their intermediate, fct/dim, and mart models. Data team members don’t need to worry about upholding standards in downstream models because this is already done when the data is first ingested.
+
+### Deeper understanding of data
+
+By first wrangling or cleaning data, you get to learn about the data’s intricacies in the process. Though manual, this process allows you to find issues in the data and understand them deeply before using them in downstream processes. This minimzes potential problems that can go unnoticed because you’ve already explored and validated the datasets. It also helps you understand how tables can be joined together downstream.
+Additionally, this initial data exploration and transformation helps you collaborate better with [backend application developers](https://docs.getdbt.com/blog/when-backend-devs-spark-joy) or data engineers to work on formatting the raw data in a format that is most appropriate for analytics work.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md:115]:
```suggestion

Additionally, this initial data exploration and transformation helps you collaborate better with [backend application developers](https://docs.getdbt.com/blog/when-backend-devs-spark-joy) or data engineers to work on formatting the raw data in a format that is most appropriate for analytics work.
```

File: website/docs/terms/data-wrangling.md [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+
+### SQL cleaning
+
+- `CAST( <column_name> AS date)` is commonly used to cast values in a column to a date datatype, or whatever other datatype you specify.
+
+- `CONVERT_TZ()` can be used to convert values in a column to a specific timezone.
+
+- `LOWER()`/`UPPER() `is used to capitalize or lowercase string values.
+
+`TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
`TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```
```suggestion
- `TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```

File: website/docs/terms/data-wrangling.md [hunk 24]
```diff
@@ -0,0 +1,157 @@
...
+
+`REPLACE()` replaces a specified character in column values.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `REPLACE()` replaces a specified character in column values.
```

File: website/docs/terms/data-wrangling.md [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+
+You can also use custom built macros, such as those from a free dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL. 
+
+### Enriching
+
+Enriching data using SQL can often involve the use of functions, such as:
+
+- CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements. 
+`IFNULL()` replaces any NULL values in a column with whatever value you specify.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `IFNULL()` replaces any NULL values in a column with whatever value you specify.
```

File: website/docs/terms/data-wrangling.md [hunk 26]
```diff
@@ -0,0 +1,157 @@
...
+`COALESCE()` returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `COALESCE()` returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```

File: website/docs/terms/data-wrangling.md [hunk 27]
```diff
@@ -0,0 +1,159 @@
...
+With the new capability of [Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [#beta-feedback-python-models channel](https://www.getdbt.com/community/join-the-community/) and join in the [GitHub discussions](https://github.com/dbt-labs/dbt-core/discussions/5261) happening around it.
+:::
+
+### Validating
+
+Validating data consists of comparing your cleaned data to the original raw data table in order to confirm no important data values were lost in the wrangling process. Rather than cleaning, you aren’t changing the resultant dataset here, instead you are comparing it to the original. With validation, you can use the same data discovery queries you use on the original dataset, on the resultant dataset, to ensure counts of values, non-nulls, and non-duplicates match. The goal is to have the same summary statistics as the original when certain filters are applied. For example, you want to have the same number of primary keys in the data set after wrangling as you did before.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
Validating data is the process of ensuring that the changes you just made to a dataset during your transformation are accurate. At this stage, you may be asking yourself:
- Are the primary keys still unique? Are there the same number of primary keys in this transformed table than in my upstream sources?
- Has the relationship with the upstream table(s) changed at all, or is it still 1-1? If not, is that expected behavior?
- Has the distribution of column values changed at all? Are column values even correct?
- Did I select the correct columns I want present at this stage?

To answer these questions, you'll likely find yourself looking for and counting nulls, rows, duplicates, and primary keys. You'll likely reference upstream models regularly in this phase to ensure your transformation code is accurate and performing what you intended it to do. 

Validation is always a little manual, but [dbt tests, macros, and packages](#validating-1) can help make your data validation a little easier 😉 .
```

File: website/docs/terms/data-wrangling.md [hunk 28]
```diff
@@ -0,0 +1,167 @@
...
+:::
+
+### Enriching
+
+Enriching your data means enhancing it by supplementing incomplete or missing data. This could involve basic case or coalesce statements that use an already existing column in order to produce a new column. It could also look like joining an already existing date column with a date table that contains more extensive information about a certain date. Keep in mind that you don’t want to go overboard with enriching or joining here—you only want to add what will be repeatedly used in modeling and analysis.
+
+:::tip Python for enrichment?
+With the new capability of [Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [#beta-feedback-python-models channel](https://www.getdbt.com/community/join-the-community/) and join in the [GitHub discussions](https://github.com/dbt-labs/dbt-core/discussions/5261) happening around it.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
With the new capability of [Python in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models), will folks start using Python to help enrich their data? Only time will tell, but we’re eager to hear how you want to be using Python in dbt. Please join the [#dbt-core-python-models channel](https://www.getdbt.com/community/join-the-community/) to join in on the discussions happening around them.
```

File: website/docs/terms/data-wrangling.md [hunk 29]
```diff
@@ -0,0 +1,167 @@
...
+### Potential performance improvements on downstream models
+
+Lastly, data wrangling allows for potential improvements in performance in downstream models. Because you’ve cleaned the data and potentially removed duplicates and null values, models will be quicker to run.
+## Data wrangling in SQL
+
+SQL is the most common language for data wrangling. While you can wrangle data using other languages, such as Python, SQL is the most common (and straightforward) language used for data wrangling and transformation in relational databases. Let’s look at some of the most common SQL functions for each of the data wrangling steps. 
+
+### SQL cleaning
+
+- `CAST( <column_name> AS date)` is commonly used to cast values in a column to a date datatype, or whatever other datatype you specify.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `CAST( <column_name> AS <data_type>)` is commonly used to cast values in a column to a specified data type.
```

File: website/docs/terms/data-wrangling.md [hunk 30]
```diff
@@ -0,0 +1,167 @@
...
+
+- `CONVERT_TZ()` can be used to convert values in a column to a specific timezone.
+
+- `LOWER()`/`UPPER() `is used to capitalize or lowercase string values.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `LOWER()`/`UPPER()` is used to capitalize or lowercase string values.
```

File: website/docs/terms/data-wrangling.md [hunk 31]
```diff
@@ -0,0 +1,167 @@
...
+
+- `TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
+
+- `REPLACE()` replaces a specified character in column values.
+
+You can also use custom built macros, such as those from a free dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
You can also use custom built macros, such as those from a dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL. 
```

File: website/docs/terms/data-wrangling.md [hunk 32]
```diff
@@ -0,0 +1,167 @@
...
+- `REPLACE()` replaces a specified character in column values.
+
+You can also use custom built macros, such as those from a dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL. 
+
+### Enriching
+
+Enriching data using SQL can often involve the use of functions, such as:
+
+- CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements. 
+- `IFNULL()` replaces any NULL values in a column with whatever value you specify.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `IFNULL` replaces any null values in a column with whatever value you specify.
```

File: website/docs/terms/data-wrangling.md [hunk 33]
```diff
@@ -0,0 +1,167 @@
...
+### Potential performance improvements on downstream models
+
+Lastly, data wrangling allows for potential improvements in performance in downstream models. Because you’ve cleaned the data and potentially removed duplicates and null values, models will be quicker to run.
+## Data wrangling in SQL
+
+SQL is the most common language for data wrangling. While you can wrangle data using other languages, such as Python, SQL is the most common (and straightforward) language used for data wrangling and transformation in relational databases. Let’s look at some of the most common SQL functions for each of the data wrangling steps. 
+
+### SQL cleaning
+
+- `CAST( <column_name> AS <data_type>)` is commonly used to cast values in a column to a specified data type.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `CAST` is commonly used to cast values in a column to a specified data type.
```

File: website/docs/terms/data-wrangling.md [hunk 34]
```diff
@@ -0,0 +1,167 @@
...
+
+- `CONVERT_TZ()` can be used to convert values in a column to a specific timezone.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `CONVERT_TZ` can be used to convert values in a column to a specific timezone.
```

File: website/docs/terms/data-wrangling.md [hunk 35]
```diff
@@ -0,0 +1,167 @@
...
+
+- `LOWER()`/`UPPER()` is used to capitalize or lowercase string values.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `LOWER`/`UPPER` is used to capitalize or lowercase string values.
```

File: website/docs/terms/data-wrangling.md [hunk 36]
```diff
@@ -0,0 +1,167 @@
...
+
+- `TRIM()` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `TRIM` can remove leading or trailing characters in strings, making string functions easier to use downstream or more consistent across tables.
```

File: website/docs/terms/data-wrangling.md [hunk 37]
```diff
@@ -0,0 +1,167 @@
...
+
+- `REPLACE()` replaces a specified character in column values.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `REPLACE` replaces a specified character in column values.
```

File: website/docs/terms/data-wrangling.md [hunk 38]
```diff
@@ -0,0 +1,167 @@
...
+
+You can also use custom built macros, such as those from a dbt package called [re_data](https://hub.getdbt.com/re-data/re_data/latest/), to clean columns using SQL. 
+
+### Enriching
+
+Enriching data using SQL can often involve the use of functions, such as:
+
+- CASE statements allow you to replace values using  “when-then” statements. They end with an “else” statement to catch the values that don’t fall in any of the “when-then” statements. 
+- `IFNULL()` replaces any NULL values in a column with whatever value you specify.
+- `COALESCE()` returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
- `COALESCE` returns the first non-null value from a list or column that you give it. This function is useful for replacing null values with one that you specify or coalescing multiple column values together.
```

File: website/docs/terms/data-wrangling.md [hunk 39]
```diff
@@ -0,0 +1,167 @@
...
+
+For nested data types such as JSON, you’ll want to check out the JSON parsing and extraction function of your data warehouse to help work with this data.
+
+### Validating
+
+dbt offers [generic tests](https://docs.getdbt.com/docs/building-a-dbt-project/tests#more-generic-tests) in every dbt project that allows you to validate accepted, unique, and null values. They also allow you to validate the relationships between tables and that the primary key is unique.
+
+If you can’t find what you need with the generic tests, you can download an additional dbt testing package called [dbt_expectations](https://hub.getdbt.com/calogica/dbt_expectations/0.1.2/) that dives even deeper into how you can test the values in your columns. This package has tests like expect_column_values_to_be_in_type_list, expect_column_values_to_be_between, and expect_column_value_lengths_to_equal.
```
Comment (KiraFuruichi) [inline website/docs/terms/data-wrangling.md]:
```suggestion
If you can’t find what you need with the generic tests, you can download an additional dbt testing package called [dbt_expectations](https://hub.getdbt.com/calogica/dbt_expectations/0.1.2/) that dives even deeper into how you can test the values in your columns. This package has useful tests like `expect_column_values_to_be_in_type_list`, `expect_column_values_to_be_between`, and `expect_column_value_lengths_to_equal`. 
```

**Review by KiraFuruichi — DISMISSED** [review]

**Review by KiraFuruichi — APPROVED** [review]
